Python/specialize.c

```diff
@@ -594,7 +595,9 @@ analyze_descriptor(PyTypeObject *type, PyObject *name, PyObject **descr, int sto
                 getattribute != interp->callable_cache.object__getattribute__;
             has_getattr = _PyType_Lookup(type, &_Py_ID(__getattr__)) != NULL;
             if (has_custom_getattribute) {
-                if (!has_getattr && Py_IS_TYPE(getattribute, &PyFunction_Type)) {
+                if (getattro_slot == _Py_slot_tp_getattro &&
```

Comment:
Why this change?

Reply:
An assertion about which slot is present during specialisation was failing in one of the buildbots. Apparently there are cases where the other slot (which potentially calls `__getattr__`) still stuck around even with no `__getattr__`. This may mean the slot had not been called yet (seems strange considering specialised code is supposed to be hot).

Reply:
Which assertion?
Does that mean the assertion is incorrect, or that this test was?

Reply:
This one https://github.com/python/cpython/pull/93988/files#diff-22e653d878778ca28317261f7826545288d02fd9980cf485d0329f28a34993f2R834.

Reply:
Strictly, that assert is wrong, in the sense that it's OK if `type->tp_getattro == _Py_slot_tp_getattr_hook` provided that there it would convert itself to `_Py_slot_tp_getattro`.

Having said that, it is probably easier to reason about if we assert `type->tp_getattro == _Py_slot_tp_getattro`, and the impact on performance of the extra test will be negligible.